src/beacon.cpp

```diff
@@ -61,3 +61,12 @@ std::string GetStoredBeaconPublicKey(const std::string& cpid)
 {
     return GetArgument("publickey" + cpid + GetNetSuffix(), "");
 }
+
+void UpdateBeaconKeysInMemory(
+        const std::string &cpid,
+        const std::string &pubKey,
+        const std::string &privKey)
+{
+    mapArgs["-publickey" + cpid + GetNetSuffix()] = pubKey;
```

Comment:
cud be stored incase someone needs to restore a backup and maybe then it affects the cache after the fact. but will do the SetArgumet stufff first as well as the name change :) then will move to the cache